### ISSUE TITLE
Unable to parse .env files with escaped newlines in values

### ISSUE THREAD

**Issue by liamdawson** [casey/just#468]
### Steps

```shell
echo -e '#!/usr/bin/env just --justfile\n\ndb:\n  echo hello world' > just
chmod +x just
echo 'FOO="bar\nbaz"' >> .env
./just db
```

```shell
$ cat just
#!/usr/bin/env just --justfile

db:
  echo hello world

$ cat .env
FOO="bar\nbaz"

```

#### Expected
```
hello world
```

#### Actual

```
error: Failed to load .env: Error parsing line: '"bar\nbaz"'
```

### Details
`just --version`: just v0.4.4

**Comment by casey** [comment]
Thank you for the comprehensive issue report!

Just uses dotenv-rs/dotenv to parse .env files, and it looks like this is an issue there. Fortunately, I checked on current master, and it's fixed. (I think by this PR https://github.com/dotenv-rs/dotenv/pull/24.) So it's just a matter of waiting until dotenv creates a new release. I'm going to leave this issue open, and will close it when they release a new version and I can update the dependency.

**Comment by liamdawson** [comment]
It's been a month and a bit (with no response) since I asked about cutting a release. Any chance to pin to an upstream git commit instead, so we can use the fix?

**Comment by casey** [comment]
Sorry that this hasn't gotten attention. I just pinged the issue, and I'd like to give it a day or two to see if upstream will release a fix. I want to avoid creating and depending on a fork, which will prevent cutting a new release of `just` without also publishing the fork to crates.io.

**Comment by liamdawson** [comment]
I think https://crates.io/crates/dotenv/0.15.0 has the fix now 😄

**Comment by casey** [comment]
I just merged #501, which bumps to the latest version of dotenv. I haven't cut a new release yet, since #492 will land soon, and I was planning on cutting a release after that. Let me know if you need it on crates.io though, and I'll happily cut a release earlier.

**Comment by liamdawson** [comment]
Any chance I could convince you to drop a patch-level release for this fix, considering #492 is still going through feedback cycles? 🙂

**Comment by casey** [comment]
You bet! I just cut v0.4.5, which should be live on crates.io.
